devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -1428,17 +1619,40 @@ void WholeBodyDynamicsDevice::readContactPoints()
         {
             if (contactsReadFromSkin.empty())
             {
-
-                for(size_t subModel = 0; subModel < nrOfSubModels; subModel++)
+                if(!overrideContactFramesSelected)
```

Comment:
the multiple if...else logic in this function is very confusing and difficult to follow. It becomes even more complex with addition of our new feature. If it's not too much work, could you please add the logic sequence as a short comment at the beginning of this function?

Reply:
unnecessary if...else conditions are removed in https://github.com/robotology/whole-body-estimators/pull/48/commits/edcd509dad868124ec98e627ecd260af425c8ca1